hostsfile/src/lib.rs

```diff
@@ -284,6 +284,41 @@ impl HostsBuilder {
     fn write_and_swap(temp_path: &Path, hosts_path: &Path, contents: &[u8]) -> io::Result<()> {
         // Copy the file we plan on modifying so its permissions and metadata are preserved.
         std::fs::copy(hosts_path, temp_path)?;
+
+        #[cfg(feature = "selinux")]
+        if selinux::current_mode() != selinux::SELinuxMode::NotRunning {
+            log::debug!("SELinux is running; copying context");
+            use selinux::SecurityContext;
+
+            match SecurityContext::of_path(hosts_path, false, false) {
+                Ok(Some(context)) => {
+                    log::debug!(
+                        "{} context is {:?}",
+                        hosts_path.display(),
+                        context.to_c_string()
+                    );
+                    if let Err(err) = context.set_for_path(temp_path, false, false) {
+                        log::warn!(
+                            "Failed to set selinux context of {} to {:?}: {:?}",
+                            temp_path.display(),
+                            context.to_c_string(),
+                            err
+                        );
```

Comment:
I like when warnings like this also state the consequence. Here we could say something like "selinux context of {} was not preserved over atomic write"..